numpy/core/src/umath/ufunc_object.c

```diff
@@ -575,13 +631,59 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
     int nout = ufunc->nout;
     int nop = ufunc->nargs;
     PyObject *obj, *context;
-    PyObject *str_key_obj = NULL;
     const char *ufunc_name = ufunc_get_name_cstr(ufunc);
     int type_num;
 
     int any_flexible = 0, any_object = 0, any_flexible_userloops = 0;
-    int has_sig = 0;
 
+    PyArray_Descr *dtype = NULL;
+    PyObject *sig = NULL;
+    typedef int (*converter)(PyObject *, void *);
+    typedef struct {
+        converter convert;
+        PyObject *str;
+        void *output;
+    } ufunc_kwarg;
+    /*
+     * Keyword definitions
+     *
+     * Note: if any changes are made, one also needs to change
+     * the initalization of ->str below as well as that of ->output
+     * in the kwds loop.
+     */
+    static ufunc_kwarg ufunc_kwargs[] = {
+        {NULL, NULL, NULL},                         /* out; no converter, special-cased */
+        {(converter)_parse_wheremask, NULL, NULL},  /* where */
+        {(converter)_copy_value, NULL, NULL},       /* axes */
+        {(converter)_copy_value, NULL, NULL},       /* axis */
+        {(converter)_parse_keepdims, NULL, NULL},   /* keepdims */
+        {NULL, NULL, NULL},                         /* casting; converter inserted below */
+        {NULL, NULL, NULL},                         /* order; converter inserted below */
+        {NULL, NULL, NULL},                         /* dtype; converter inserted below */
+        {(converter)_parse_subok, NULL, NULL},      /* subok */
+        {(converter)_copy_value, NULL, NULL},       /* signature */
+        {(converter)_copy_value, NULL, NULL},       /* sig */
+        {(converter)_copy_value, NULL, NULL},       /* extobj */
+        {NULL, NULL, NULL}  /* sentinel */
+    };
+    ufunc_kwarg *kwarg = ufunc_kwargs;
+    if (kwarg->str == NULL) {  /* fill on first call */
```

Comment:
Worth adding that this is because the strings are not interned yet during static initialization.

You could store `&npy_um_str_out` etc into the array instead, which would work fine during static initialization

Reply:
OK, better indeed.